lib/ansible/modules/messaging/kafka_topics.py

```diff
@@ -0,0 +1,384 @@
+        if len(self.topic) > 1:
+                msg = 'alter action supports only 1 topic'
+                return self.module.fail_json(msg)
+        else:
+            cmd = ('%s --delete --zookeeper %s '
+                   '--topic %s --if-exists') % (self.executable,
+                                                self.zookeeper,
+                                                ''.join(self.topic))
+
+        try:
+            env = ''
+            if self.jaas_auth_file:
+                env = {'KAFKA_OPTS': self.kafka_env_opts}
+
+            return self.module.run_command(cmd, environ_update=env)
+
+        except KafkaError as exc:
+            self.module.fail_json(msg=to_native(exc))
+
+
+def main():
+    argument_spec = dict(
+        action=dict(default='list',
+                    choices=['list', 'describe', 'alter',
+                             'create', 'delete'],
+                    type='str'),
+        topic=dict(type='list'),
+        config=dict(type='dict'),
+        zookeeper=dict(required=True, type='list'),
+        replication_factor=dict(type='int'),
+        partitions=dict(type='int'),
+        executable=dict(default='/usr/bin/kafka-topics',
+                        type='path'),
+        jaas_auth_file=dict(type='path'),
+    )
+
+    required_if = [
+        ['action', 'alter', ['topic', 'partitions']],
+        ['action', 'create',
+         ['topic', 'replication_factor', 'partitions']
+         ],
+        ['action', 'delete', ['topic']],
+    ]
+
+    module = AnsibleModule(
+        argument_spec=argument_spec,
+        required_if=required_if,
+        supports_check_mode=True,
+    )
+
+    changed = False
+
+    if not os.path.isfile(module.params['executable']):
+        module.fail_json(msg='%s not found.' % (module.params['executable']))
+
+    if module.params['jaas_auth_file']:
+        if not os.path.isfile(module.params['jaas_auth_file']):
+            module.fail_json(msg='%s not found.' % (module.params['jaas_auth_file']))
+
+    try:
+        kt = KafkaTopics(module)
+
+        action = module.params['action']
+        if not module.check_mode:
+            if module.params['action'] == 'list':
+                (rc, out, err) = kt.get()
+            elif module.params['action'] == 'describe':
+                (rc, out, err) = kt.describe()
+            elif module.params['action'] == 'alter':
+                (rc, out, err) = kt.alter()
+                if re.search('Adding partitions succeeded!', out):
+                    changed = True
+            elif module.params['action'] == 'create':
+                (rc, out, err) = kt.create()
+                if re.search('^Created topic', out):
+                    changed = True
+            elif module.params['action'] == 'delete':
+                (rc, out, err) = kt.delete()
+                if re.search('^Topic [a-zA-Z0-9_\-]+ is marked for deletion.', out):
+                    changed = True
+
+            result = {
```

Comment:
In case of failure, `return['failed']` should be `True`: new modules must not use `rc` instead of `failed` (See [dev_guide](http://docs.ansible.com/ansible/devel/dev_guide/developing_modules_best_practices.html#conventions-best-practices-and-pitfalls) and 2a041d10d26d5746132061da4b874f33cb73ffb0).